### ISSUE TITLE
Not all distros have systemd

### ISSUE THREAD

**Issue by Keridos** [searx/searx-docker#9]
Fixed in https://github.com/searxng/searxng-docker if anyone is interested.

-----------

I know you can write your own service files for any service daemon. But considering that not having systemd makes the "within 5 minutes" pretty much impossible. Why not build an official image based on alpine with each commit from the searx repository? That would be literally a 1 minute setup.

**Comment by jfdhuiz** [comment]
Not all Docker systems run Linux either. I'd have liked to have read the systemd dependancy a bit higher e.g. say "setup is 5 minutes (with systemd)".

**Comment by dalf** [comment]
What are you using ? SystemV init scripts ?

Note: the systemd service doesn't do much: just call start.sh and stop.sh.

>  Why not build an official image based on alpine with each commit from the searx repository?

There is this image : https://hub.docker.com/r/searx/searx
Unfortunately this is not the whole thing: you need filtron, morty and a configuration for your reverse proxy. That's the purpose of this project using Docker. Without Docker, there is work in progress: https://github.com/asciimoo/searx/pull/1803

**Comment by jfdhuiz** [comment]
> What are you using ? SystemV init scripts ?

Yeah, this was Docker running on a Synology machine running DSM 6.2.

I'd really like to run the whole thing within Docker on that machine.

I'm not sure on the reverse proxy part. I don't run any of these as of now. It gets a bit messy with a lot of services and increases attack surface, but OTOH the machine isn't reachable outside of VPN and LAN. Ie. this is going to be a private searx instance. So no need for Let's Encrypt, TLS, and such.

**Comment by dalf** [comment]
Which architecture is this ? amd64, x86, arm ?

For now, it's only works for the amd64 platform, but most probably you only need searx:
* either install it the usual way ( https://asciimoo.github.io/searx/admin/installation.html )
* either build the docker image by yourself for arm.
* either make a PR to support arm architecture.

If your NAS has an amd64 architecture, then can you create a task to run start.sh in the scheduler (inside the DSM UI) ?

**Comment by jfdhuiz** [comment]
This one is a AMD64. Most Synology these days are.

I'm using a script called start.sh which contains all the Docker services I want. I start that manually after a reboot because I first need to manually enable the FDE. I don't know if it works for this one until I've rebooted, but I assume it is going to work (I just removed the Caddy stuff from docker-compose.yaml because I don't need it).

So from my PoV this issue is solved. It is just that I'd like some kind of heads up on top of the README that it is for Linux, not macOS/Windows. Plus I guess a notice that systemd is optional, or for people who use systemd (macOS/Windows users certainly don't).

**Comment by unixfox** [comment]
The thing is in reality searx-docker doesn't need to have systemd installed, in fact I'm using it on Alpine which has openrc instead of systemd.
To remove any dependency on systemd the docker-compose has to be refactored to work with caddy-docker-proxy (like described in #23) thus removing any hardcoded ip like here: https://github.com/searx/searx-docker/blob/master/docker-compose.yaml#L40.

**Comment by unixfox** [comment]
Fixed in https://github.com/searxng/searxng-docker if anyone is interested.
